src/migrateDashboard.ts

```diff
@@ -64,14 +107,42 @@ export function migrateDashboard(
       page.layout = _convertFromLegacyLayout(legacyLayout);
       _flattenLayout(page.layout);
     }
-    pages[`p-${index}`] = page;
+
+    pages[pageKey] = page;
   });
 
-  return {
+  const dashboard: DashboardState<"serialized"> = {
     name: legacyDashboardState.name,
     pages,
     pagesOrder: _range(body.pages.length).map((i) => `p-${i}`),
     filters: Object.values(body.filters || {}).flat(),
     queryContext: _migrateContextValues(body.contextValues),
   };
+
+  const deserializedDashboard = deserializeDashboardState(dashboard);
+
+  const dashboardWithWidgetsRemoved = _reduce(
```

Comment:
That would be ideal, but this is not the signature of `removeWidget`:
```
removeWidget({
    dashboardState,
    pageKey,
    leafKey,
    layoutPath,
});
```
And since `removeWidget` can modify the layout of a given page, then when calling several `removeWidget` in a row, one has to get the latest `dashboardState` at each iteration (rather than the initial `dashboardState` before any removal).
So typically I fell this is a good use case for reduce: you want to feed the intermediate result to the next iteration step.

Or do you have another idea in mind ?

Reply:
I am pushing an attempt. Let me know what you think.